app/models/user.rb

```diff
@@ -18,34 +18,26 @@ class User < ActiveRecord::Base
   enum role: [:auditor, :accountant, :manager, :admin]
 
   ## Callbacks
-  after_create :make_admin!
+  after_commit :make_admin!
 
   # Override Devise::Confirmable#after_confirmation
   def after_confirmation
-    send_reset_password_instructions unless admin? && admin_id == id
+    send_reset_password_instructions unless admin?
   end
 
   def colleagues
-    if company.present?
-      company.users.where.not(role: User.roles[:admin])
-    else
-      User.where("role <> ?",User.roles[:admin]).where(admin_id: id).where.not(id: id)
-    end
+    company.users.where.not(role: User.roles[:admin])
   end
 
   def make_admin!
-    if role.nil? && admin_id.nil?
-      user = User.find_by(admin_id: id)
-      unless user.present? && user.admin?
-        self.admin_id = id
-        self.role = self.class.roles[:admin]
-        self.save
-      end
+    unless role.present?
+      self.role = self.class.roles[:admin]
+      self.save
     end
   end
 
   def other_roles
-    ((admin_id.present?) && (admin_id != id))
+    admin?
```

Comment:
You are not doing additional thing other than checking `admin?` then can we not use that method itself?